File: Caper/CaperSharp/BowtieMappingEngine.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace CaperSharp
{
  class BowtieMappingEngine : MappingEngine
  {
    public BowtieMappingEngine( string aPath, Sequences aReferenceGenome )
      : base( aPath, aReferenceGenome )
    {
    }
    
    public override string GetSequence( string aLine )
    {
      return GetLinePieces( aLine )[ 4 ];
    }

    public override int GetIndex( string aLine )
    {
      return GetNormalizedIndex( GetLinePieces( aLine )[ 3 ] );
    }

    private int GetNormalizedIndex( string aValue )
    {
      return System.Convert.ToInt32( aValue );
    }

    public override string GetContigIdent( string aLine )
    {
      return GetLinePieces( aLine )[ 2 ];
    }
  }
}

File: Caper/Caper/BowtieMappingEngine.cpp
#include "BowtieMappingEngine.h"

BowtieMappingEngine::BowtieMappingEngine(string & aPath, Sequences & aGenome) 
: MappingEngine(aPath, aGenome), BowtieMappingUtilities()
{
}

string BowtieMappingEngine::GetContigIdent( string & aLine )
{
  return BowtieMappingUtilities::GetContigIdent(aLine);
}

int BowtieMappingEngine::GetIndex( string & aLine )
{
  return BowtieMappingUtilities::GetIndex(aLine);
}

string BowtieMappingEngine::GetName( string & aLine )
{
  return BowtieMappingUtilities::GetName(aLine);
}

string BowtieMappingEngine::GetSequence( string & aLine )
{
  return BowtieMappingUtilities::GetSequence(aLine);
}
